packages/runtime-common/index-writer.ts

```diff
@@ -250,6 +255,56 @@ export class Batch {
     return parseInt(total);
   }
 
+  private async updateRealmMeta() {
+    let results = await this.query([
+      `SELECT count(i.url) as total, i.display_names->>0 as display_name, i.types->>0 as code_ref
+       FROM boxel_index as i
+       INNER JOIN realm_versions r ON i.realm_url = r.realm_url
+          WHERE`,
+      ...every([
+        ['i.realm_url =', param(this.realmURL.href)],
+        ['i.type = ', param('instance')],
+        ['i.types IS NOT NULL'],
+        realmVersionExpression({ useWorkInProgressIndex: true }),
```

Comment:
this is fine, you could have also used `withMaxVersion: this.realmVersion` too. I don't really have a preference....